### PR TITLE
fix: stabilize head metadata responses and heal tests

### PR DESCRIPTION
**Pull request by LoganZ2**
Type of Change
- [ ] New Feature
- [x] Bug Fix
- [ ] Documentation
- [ ] Performance Improvement
- [x] Test/CI
- [ ] Refactor
- [ ] Other:

Related Issues
#1295 

Summary of Changes
Ensure HeadObject returns standard Content-Type and Content-Disposition headers derived from stored metadata while keeping x-amz-meta-* for backward compatibility across SDKs. Stabilize heal integration tests by validating healed object readability instead of relying on transient on-disk part paths. Guard concurrent request tracking against underflow in edge cases to keep metrics consistent.

Checklist
- [x] I have read and followed the CONTRIBUTING.md guidelines
- [x] Passed make pre-commit
- [x] Added/updated necessary tests
- [ ] Documentation updated (if needed)
- [ ] CI/CD passed (if applicable)

Impact
- [ ] Breaking change (compatibility)
- [ ] Requires doc/config/deployment update
- [ ] Other impact:

### CODE REVIEW

File: crates/heal/tests/heal_integration_test.rs [hunk 1]
```diff
@@ -320,11 +331,14 @@ mod serial_tests {
         let heal_manager = HealManager::new(heal_storage.clone(), Some(cfg));
         heal_manager.start().await.unwrap();
 
+        let (_result, error) = heal_storage.heal_format(false).await.expect("Failed to heal format");
+        assert!(error.is_none(), "Heal format returned error: {error:?}");
```
[the remark this answers is not in the document]
Comment (houseme) [reply crates/heal/tests/heal_integration_test.rs]: @LoganZ2

File: rustfs/src/storage/concurrency.rs [hunk 2]
```diff
@@ -426,8 +426,8 @@ impl Drop for GetObjectGuard {
     /// This ensures accurate tracking even in error/panic scenarios, as Drop
     /// is called during stack unwinding (unless explicitly forgotten).
     fn drop(&mut self) {
-        // Decrement concurrent request counter
-        ACTIVE_GET_REQUESTS.fetch_sub(1, Ordering::Relaxed);
+        // Decrement concurrent request counter without underflow
+        let _ = ACTIVE_GET_REQUESTS.fetch_update(Ordering::Relaxed, Ordering::Relaxed, |current| current.checked_sub(1));
```
[the remark this answers is not in the document]
Comment (houseme) [reply rustfs/src/storage/concurrency.rs]: @LoganZ2

File: rustfs/src/storage/ecfs.rs [hunk 3]
```diff
@@ -3417,6 +3417,18 @@ impl S3 for FS {
         // takes precedence for these read operations.
         let mut response = wrap_response_with_cors(&bucket, &req.method, &req.headers, output).await;
 
+        if let Some(content_disposition) = metadata_map.get("content-disposition")
+            && let Ok(header_value) = HeaderValue::from_str(content_disposition)
+        {
+            response.headers.insert(http::header::CONTENT_DISPOSITION, header_value);
```
[the remark this answers is not in the document]
Comment (houseme) [reply rustfs/src/storage/ecfs.rs]: @LoganZ2
